Ignore reserved VLANs 1002-1005 for state overridden

jiholland opened the pull request:
##### SUMMARY
Ignore VLAN 1002-1005 for state overridden

##### ISSUE TYPE
- Bugfix Pull Request

#1182

jiholland committed:
fix: ignore reserved vlans 1002-1005

jiholland committed:
docs: do not delete reserved vlans 1, 1002-1005

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

KB-perByte committed:
Merge branch 'main' into fix

jiholland committed:
fix: ignore reserved VLANs 1002-1005 for state overridden

jiholland committed:
fix: include vlan 1002-1005 when gathering vlan facts

jiholland committed:
fix: remove changelog

Ruchip16 commented on plugins/module_utils/network/ios/config/vlans/vlans.py:
```diff
@@ -118,6 +118,9 @@ def generate_commands(self, conf_want, conf_have, resource=None):
             excluded_vlans = {k: v for k, v in iteritems(haved) if k not in wantd or not wantd}
             haved = {k: v for k, v in iteritems(haved) if k in wantd or not wantd}
             for k, have in iteritems(excluded_vlans):
+                # Skip reserved VLANs 1002–1005
```
why are we skipping these?

jiholland replied:
Because they can’t be deleted, which breaks idempotency for state overridden.

Ruchip16 replied:
integration/unit tests maybe

KB-perByte replied:
Hey @jiholland I am taking a look at this.
Will update shortly

jiholland replied:
Thanks @KB-perByte . Idempotency with state overridden is still an issue, and it would be great if it could be resolved. It does not have to be this exact fix, but some solution to the problem would be nice.

jiholland replied:
Any updates?

KB-perByte committed:
Update integration-simple.yml

KB-perByte committed:
Merge branch 'main' into fix

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

AAYUSH2091 committed:
Fix cli_conf integration test (#1228)

* minor_changes

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Update commit_conf.yaml

* Update commit_conf.yaml

* Update commit_conf.yaml

* Update tests/integration/targets/ios_cliconf/tests/common/commit_conf.yaml

* added_pexpect

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Update cli_test.yml

---------

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>
Co-authored-by: Sagar Paul <sagpaul@redhat.com>

AAYUSH2091 committed:
Fix parsing of static routes with interface and distance in gathered state (#1227)

* minor_fix

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Update static_routes.yml

* Update test_ios_static_routes.py

* fixes_integration_test

* reverting_tests_changes

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

---------

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

AAYUSH2091 committed:
Fix idempotency for multiple route-targets (#1226)

* minor_changes

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Update vrf_global.py

* fixes_lint_errors

* fixes_lint_errors

* Update .ansible-lint

* Update .ansible-lint

* fix_lint_errors

* fixing_lint_erros

* reverting_the_changes_of_lint_file

---------

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

KB-perByte committed:
Update integration-simple.yml

KB-perByte committed:
Update integration-simple.yml

KB-perByte committed:
enable workflow

KB-perByte committed:
Update names of GHA

ollylawrence committed:
Add chassis_id LLDP fact to net_neighbours fact. (#1229)

* Added in chassis_id to net neighbours

* added test for chassis_id in lldp neighbours.

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* added changelog fragment

---------

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>
Co-authored-by: Sagar Paul <sagpaul@redhat.com>

KB-perByte committed:
Names and fix precommit

AAYUSH2091 committed:
Fix ios bgp address family parsing for ipv6 (#1235)

* fix_ios_bgp_address_family

* minor_fix

* minor_fix

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Update commit_conf.yaml

* added_integration_tests

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Update bgp_address_family.yml

* Update bgp_address_family.py

---------

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

komaldesai13 committed:
Added prompt and answer functionalities  to ios config module (#1232)

* Added prompt and answer funtinalities to ios config module

* added robustness to code

* added description about line , prompt and answer

* added robustness to code

* removed traceback

* added unit tests

* added integration tests

* added changelog

* addressed review comments

* fixed sanity

* fixed sanity

* fixed sanity

* fixed sanity

* fixed sanity

* fixed test

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Update prompt.yaml

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

---------

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>
Co-authored-by: Sagar Paul <sagpaul@redhat.com>

KB-perByte committed:
[ios_hsrp_interfaces] ipv6 attributes handling issues (#1240)

* ipv6 handling issues with ios_hsrp_interface

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* fix state and defaults

* update defaults and tasks

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* only apply version if standby is present

* Update fixes

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* fix tests

* fix sanity

* Update pre-commit

---------

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

KB-perByte committed:
Prep release 11.1.0 (#1242)

jiholland committed:
docs: do not delete vlan 1002-1005 for state overridden

jiholland committed:
fix: update tests for reserved vlan 1002-1005

jiholland committed:
Merge branch 'main' into fix

jiholland committed:
fix: add comment

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

jiholland committed:
Merge branch 'main' into fix

jiholland committed:
Merge branch 'main' into fix

jiholland committed:
Merge branch 'main' into fix

jiholland committed:
Merge branch 'main' into fix

jiholland committed:
Merge branch 'main' into fix
